include/pybind11_json/pybind11_json.hpp

```diff
@@ -109,7 +109,7 @@ namespace nlohmann
                 }
                 return out;
             }
-            throw std::runtime_error("to_json not implemented for this type of object: " + obj.cast<std::string>());
+            return to_json_impl(py::repr(obj));
```

Comment:
I am not really sure about this. I wonder if this is the same behavior as when using the Python `json` module. I remember it throwing when the serialization is not possible.

Reply:
The Python `json` module throws an error for non-serializable objects:

`TypeError: Object of type module is not JSON serializable`

Reply:
ok let's keep this consistent with the Python module then.